libs/remix-ui/editor/src/lib/remix-ui-editor.tsx

```diff
@@ -756,69 +761,38 @@ export const EditorUI = (props: EditorUIProps) => {
-      contextMenuGroupId: 'gtp', // create a new grouping
-      keybindings: [
+    const executeGptGenerateDocumentationAction = (functionNode) => {
+      return {
+        id: 'generateDocumentation',
+        label: intl.formatMessage({ id: 'editor.generateDocumentation' }),
+        contextMenuOrder: 0, // choose the order
+        contextMenuGroupId: 'gtp', // create a new grouping
+        keybindings: [
         // Keybinding for Ctrl + H
-        monacoRef.current.KeyMod.CtrlCmd | monacoRef.current.KeyCode.KeyH,
-      ],
-      run: async () => {
-        const unsupportedDocTags = ['@title'] // these tags are not supported by the current docstring parser
-        const file = await props.plugin.call('fileManager', 'getCurrentFile')
-        const content = await props.plugin.call('fileManager', 'readFile', file)
-        const message = intl.formatMessage({ id: 'editor.generateDocumentationByAI' }, { content, currentFunction: currentFunction.current })
-
-        // do not stream this response
-        const pipeMessage = `Generate the documentation for the function **${currentFunction.current}**`
-        await props.plugin.call('popupPanel', 'showPopupPanel', true)
-        setTimeout(async () => {
-        // const cm = await await props.plugin.call('remixAI', 'code_explaining', message)
-          const cm = await props.plugin.call('remixAI' as any, 'chatPipe', 'solidity_answer', message, '', pipeMessage)
-          const natSpecCom = "\n" + extractNatspecComments(cm)
-          const cln = await props.plugin.call('codeParser', "getLineColumnOfNode", currenFunctionNode)
-          const range = new monacoRef.current.Range(cln.start.line, cln.start.column, cln.start.line, cln.start.column)
-          const lines = natSpecCom.split('\n')
-          const newNatSpecCom = []
-
-          for (let i = 0; i < lines.length; i++) {
-            let cont = false
-
-            for (let j = 0; j < unsupportedDocTags.length; j++) {
-              if (lines[i].includes(unsupportedDocTags[j])) {
-                cont = true
-                break
-              }
-            }
-            if (cont) {continue}
-
-            if (i <= 1) { newNatSpecCom.push(' '.repeat(cln.start.column) + lines[i].trimStart()) }
-            else { newNatSpecCom.push(' '.repeat(cln.start.column + 1) + lines[i].trimStart()) }
+          monacoRef.current.KeyMod.CtrlCmd | monacoRef.current.KeyCode.KeyH,
+        ],
+        run: async () => {
+          if (functionNode) {
+            const uri = currentFileRef.current + '-ai'
+            const content = editorModelsState[currentFileRef.current].model.getValue()
+            const query = intl.formatMessage({ id: 'editor.generateDocumentationByAI' }, { content, currentFunction: currentFunction.current })
+            const output = await props.plugin.call('remixAI', 'code_explaining', query)
+            const originalFunctionComments = extractFunctionComments(content, 0, true)
```

Comment:
you can use insead `functionNode.src.split(':')[0]` which gives you the index in the file where the function is located.